hazelcast/src/main/java/com/hazelcast/wan/impl/WanReplicationServiceImpl.java

```diff
@@ -268,4 +285,130 @@ public void removeWanEventCounters(String serviceName, String objectName) {
         // EE WAN replication not supported in OS
         return Collections.emptyList();
     }
+
+    @Override
+    public void beforeMigration(PartitionMigrationEvent event) {
+        for (DelegatingWanReplicationScheme wanReplication : wanReplications.values()) {
+            for (WanReplicationPublisher publisher : wanReplication.getPublishers()) {
+                if (publisher instanceof MigrationAwareWanReplicationPublisher) {
+                    ((MigrationAwareWanReplicationPublisher) publisher).onMigrationStart(event);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void commitMigration(PartitionMigrationEvent event) {
+        for (DelegatingWanReplicationScheme wanReplication : wanReplications.values()) {
+            for (WanReplicationPublisher publisher : wanReplication.getPublishers()) {
+                if (publisher instanceof MigrationAwareWanReplicationPublisher) {
+                    ((MigrationAwareWanReplicationPublisher) publisher).onMigrationCommit(event);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void rollbackMigration(PartitionMigrationEvent event) {
+        for (DelegatingWanReplicationScheme wanReplication : wanReplications.values()) {
+            for (WanReplicationPublisher publisher : wanReplication.getPublishers()) {
+                if (publisher instanceof MigrationAwareWanReplicationPublisher) {
+                    ((MigrationAwareWanReplicationPublisher) publisher).onMigrationRollback(event);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void init(NodeEngine nodeEngine, Properties properties) {
+        // NOP
+    }
+
+    @Override
+    public void reset() {
+        final Collection<DelegatingWanReplicationScheme> publishers = wanReplications.values();
+        for (DelegatingWanReplicationScheme publisherDelegate : publishers) {
+            for (WanReplicationPublisher publisher : publisherDelegate.getPublishers()) {
+                publisher.reset();
+            }
+        }
+    }
+
+    @Override
+    public void shutdown(boolean terminate) {
+        reset();
+    }
+
+    @Override
+    public Collection<ServiceNamespace> getAllServiceNamespaces(PartitionReplicationEvent event) {
+        if (wanReplications.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        final Set<ServiceNamespace> namespaces = new HashSet<>();
+        for (DelegatingWanReplicationScheme publisher : wanReplications.values()) {
+            publisher.collectAllServiceNamespaces(event, namespaces);
+        }
+        return namespaces;
+    }
+
+    @Override
+    public boolean isKnownServiceNamespace(ServiceNamespace namespace) {
```

Comment:
why is this for only map-service? Can you please add a short explanatory javaDoc on this method?

Reply:
On OS side, we only generate WAN events for map service. I'll add the javadoc.